hal/src/can.rs

```diff
@@ -0,0 +1,171 @@
+//! This module provides target specific integration with `mcan` crate.
```

Comment:
I think the module should have more documentation than this.

Reply:
Added few sentences. In general, it's just bootstraping code to guarantee safe usage of an actual `mcan` crate.